glue/viewers/profile/state.py

```diff
@@ -338,6 +402,13 @@ def update_profile(self, update_limits=True):
             axis_view[pix_cid.axis] = slice(None)
             axis_values = data[self.viewer_state.x_att, tuple(axis_view)]
 
+            converter = UnitConverter()
+            axis_values = converter.to_unit(self.viewer_state.reference_data,
+                                            self.viewer_state.x_att, axis_values,
+                                            self.viewer_state.x_display_unit)
```

Comment:
Something does not look quite right here. Why does this use reference data but the next line use input data? I tried changing this but I cannot make it work.

I am seeing over at https://github.com/spacetelescope/jdaviz/pull/2048 that when I add a second spectrum with GHz (first spectrum is loaded in micron and plot is showing micron), the second data plots GHz values as if they are micron without any conversion.

The inputs to custom Jdaviz converter registers as such:

```python
# Signature: to_unit(self, data, cid, values, original_units, target_units)

data = Jy_um  # But I am loading mJy_GHz
cid = World 0
values = [142758.31333333 123202.38       108358.71975904  96707.24451613   87318.19165049  79590.9180531   73120.11170732  67622.35894737   62893.52265734  58782.83490196]
original_units = um  # This is wrong, should be GHz
target_units = um
```

It is also very confusing that the custom converter takes one more arg than what is really used over here.